### PR TITLE
CDPT-71 Add request method options to SAR creation and display

### PR DESCRIPTION
**Pull request by Polsonby**
## Description
When a London Disclosure user creates a SAR, they should be able to specify how the SAR was received by MOJ

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [ ] (3) Tests passing
* [ ] (4) Branch ready to be merged (not work in progress)
* [ ] (5) No superfluous changes in diff
* [ ] (6) No TODO's without new ticket numbers
* [ ] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`
* [ ] (8) Data migration script is created if any of letter templates is changed

### Screenshots
image

image

image

image

### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CDPT-71
https://dsdmoj.atlassian.net/browse/CDPT-89

### Deployment
This PR contains a data migration - see the Readme. When it is deployed onto a namespace, run `rake data:migrate` to populate the SAR case request_method values.

### Manual testing instructions
Sign into the app as David Attenborough and create a new SAR case. You should be able to specify the request method from the options Post, Email, Web Portal and so on. 

When the case details page is displayed the correct request method should be displayed. 

When an Admin user creates a SAR case, the default settings should create a valid case using "Post" as the request method.

### CODE REVIEW

File: app/views/cases/shared/_request_method.html.slim [hunk 1]
```diff
@@ -0,0 +1,3 @@
+tr.request-method
+  th = t('common.case.request_method')
+  td = case_details.request_method.humanize
```
Comment (zheileman) [inline app/views/cases/shared/_request_method.html.slim:3]: It's best not to use the `#humanize` method, as that's not doing proper i18n, can give you unexpected results and also rely on a bad practice of coupling internal identifiers to external "human-readable" strings.

Better to do explicit translation with locales, for example:

```ruby
t(case_details.request_method, scope: 'common.case.request_method.choices')
```

I know many other places in the codebase are using `#humanize` unfortunately. Something for a future refactor...

**Review by zheileman — APPROVED** [review]
LGTM 👍 
I've added a few suggestions but approving anyways as these are not critical 🥳

File: spec/factories/case/sar_internal_reviews.rb [hunk 2]
```diff
@@ -49,6 +49,7 @@
     sequence(:name)               { |n| "#{identifier} name #{n}" }
     email                         { Faker::Internet.email(name: identifier) }
     reply_method                  { 'send_by_email' }
+    request_method                { 'post' }
```
[the remark this answers is not in the document]
Comment (Polsonby) [reply spec/factories/case/sar_internal_reviews.rb:52]: Brilliant, thanks for the suggestions, @zheileman - will take the approach you suggest for the validation, nice

File: app/models/case/sar/standard.rb [hunk 3]
```diff
@@ -91,6 +100,7 @@ def self.searchable_fields_and_ranks
 
   validates_presence_of :reply_method
   validates_presence_of :subject_type
+  validates_presence_of :request_method
```
[the remark this answers is not in the document]
Comment (Polsonby) [reply app/models/case/sar/standard.rb:103]: Nice, thanks for the suggestion

File: app/views/cases/shared/_request_method.html.slim (hunk 1 above)
Comment (Polsonby) [reply app/views/cases/shared/_request_method.html.slim:3]: Thanks, it did occur to me to wonder why we weren't using the locale for the enum value display as well but couldn't think of the solution. That's great. I think the change is outside the scope of this PR as there's a bunch of these partials that all use this approach, though

**Pull request merged**